lib/datadog/grok/src/parse_grok_rules.rs

```diff
@@ -183,13 +197,23 @@ fn parse_pattern(
     pattern = pattern.replace("(?s)", "(?m)").replace("(?-s)", "(?-m)");
 
     // compile pattern
-    let pattern = grok
-        .compile(&pattern, true)
-        .map_err(|e| Error::InvalidGrokExpression(pattern, e.to_string()))?;
+    let pattern = Arc::new(
+        Regex::new(&pattern).map_err(|e| Error::InvalidGrokExpression(pattern, e.to_string()))?,
+    );
```

Comment:
It's not clear to me why this would need to an `Arc`. I know, from a previous comment in the now deleted `grok.rs` why this is but we don't explain it here anymore. Forcing multiple CPUs to coordinate because the caller -- VRL -- happens to need Clone + Sync + Send is a pain.